www: sync w/ current nodejs.org config

rvagg opened the pull request:
Addresses #200 and a few other minor things that are on the server but are not reflected in the scripts for it.

Adds the log-collect stuff that's on there collecting logs for us from cloudflare and doing some basic filtering. There's also some stuff in there to convert the json format to nginx format but it's not being used for anything currently. The log-collect work is only partially complete since we're collecting but not using it for metrics generation yet. We still need to (a) confirm that we're successfully collecting everything with this procedure (logshare isn't the most robust tool unfortunately, or maybe it's their API, hence the sanity-check script in here), and (b) either do the json->nginx conversion for metrics, or make metrics scripts able to read the json files as they are. Either way, this PR reflects current state so is good to merge IMO.

I haven't tested this .. a bit of a stab in the dark, I'm not willing to run this against the live server.

maclover7 commented:
Hmm, maybe better to pull the `log-collect` stuff out into a different PR, and land everything else first? The rest of the changes don't look too bad.

rvagg commented:
OK, removed log-collect from here, will push that in a moment. I've added an upgrade to Node 8 in this too, see https://github.com/nodejs/nodejs.org/pull/1459 for website team's agreement on this.
